pkg/cloudprovider/providers/azure/azure_loadbalancer.go

```diff
@@ -378,15 +396,17 @@ func (az *Cloud) EnsureLoadBalancerDeleted(clusterName string, service *v1.Servi
 	return nil
 }
 
-func (az *Cloud) cleanupLoadBalancer(clusterName string, service *v1.Service, isInternalLb bool) error {
+func (az *Cloud) cleanupLoadBalancer(clusterName string, service *v1.Service, isInternalLb bool) (string, error) {
```

Comment:
I don't recommend to get serviceIPToClean up from cleanupLoadBalancer function.
Because `cleanupLoadBalancer` function will be deprecated soon. 

And also, the way it links between cleanupLoadBalancer and NSG rule cleanup, could easily result in a dangling resource issue.

Think about we cleanupLoadBalancer, and get 429 before you clean NSG rules. Then the next run will never try to clean up the NSG rules, because LB's IP already gone.

Reply:
We can't look up the IP address from v1.Service but I can break out a separate call to find the service IP.  Thanks!

Reply:
I thought we can. Won't service.Spec.LoadBalancerIP is the address?

Reply:
I wish!  *grin*  But that's only populated if the service author _specified_ a LoadBalancerIP.  If the service creator allowed Kubernetes to choose the IP, then it's only available on the service.Status.LoadBalancerStatus.Ingress... and that's not cached.  (I originally used that but I had to change service controller to cache it, and @khenidak felt that would make it hard to get the PR through.)